examples/user_interaction/client/src/App.js

```diff
@@ -29,21 +30,32 @@ class App extends React.Component {
   handleClick() {
     // Use promises to test behavior on MicroTasks.
     const promise = new Promise(resolve => {
+      // Start a child span for the setTimeout as this is the operation we want 
+      // to measure. 
+      // This span will be child of the  current root span related to the 
+      // current user interaction. Additionally, these spans should be created
+      // in the code the click handler will run.
+      const setTimeoutCustomSpan = tracing.tracer.startChildSpan({ name: 'setTimeout custom span' });
       setTimeout(function () {
         resolve();
+        // End the span as the setTimeout has finished running the callback.
+        setTimeoutCustomSpan.end();
       }, 1000);
     });
 
     promise.then(() => {
-      console.log("Resolving promise");
       this.callSleepApi();
     });
   }
 
   callSleepApi() {
     const xhr = new XMLHttpRequest();
+    // Create a child span for the XHR.
+    const callSleepApiCustomSpan = tracing.tracer.startChildSpan({ name: 'Call Sleep API' });
```

Comment:
What would you think about removing this custom span since there will already be one for the XHR?

Reply:
I would like to keep it. My thought is to show in the example that is possible to create custom spans even if those tasks already generate automatic spans. I will add some comments to clarify more this example.

Reply:
OK, I think it's fine to keep. Is there another call that does not have a custom span on it?

Reply:
Well, the call to prime numbers API does not have a custom span, as it also is an XHR, I decided to not create a custom span for that.